docs/rst/enhancements/keys/topic-keys.rst

```diff
@@ -5,49 +5,80 @@ Topic Keys
+
+* Efficient middleware infrastructure, data distribution and resource utilization as the number of entities are dramatically reduced.
+* A better history management, meaning that cached updates are managed in a more performant way for determined Quality of Service parameters.
+
+These benefits are further detailed in the :ref:`benefits_of_topic_keys` section.
 
 Background
 ^^^^^^^^^^
 *Vulcanexus* uses :ref:`Fast DDS as middleware <vulcanexus_middleware>`.
-One of the core concepts involved in the communication architecture are topics,
-which serve as the bridge that enable data exchange among nodes within a particular system.
+One of the core concepts involved in the *Vulcanexus* communication architecture is topics,
+which serve as the bridge that enables data exchange among nodes within a particular system.
 
-Topics are named communication channels that allow nodes (software modules) in a robotic system
+Topics are named communication channels that allow nodes (software modules) in a distributed system (such as a robotic application)
 to exchange messages with each other. These messages can contain various types of data, such as sensor readings,
 control commands, or status updates, enabling different parts of the robot to collaborate and share information.
 
-The following image depicts a simple scenario where two sensors publish data in different topics, and two nodes,
-the first ``Logging`` node subscribes to the Sensor 2 output topic, and the ``Controller`` node that subscribes
+The following image depicts a simple scenario where two sensors publish data in different topics, and two remote nodes subscribe to it.
+The first ``Logging`` node only subscribes to the Sensor 2 output topic, while the ``Controller`` node subscribes
 to both topics.
 
 .. image:: ../../figures/enhancements/keys/topics.gif
 
 When any new data is available in the sensors, it is transmitted through the corresponding topic and all the subscribers
 attached to that topic receive the message.
 
+In this situation, each of the sensors has a dedicated topic for transmitting its data. This suggests excessive resource usage because the middleware
+has to create and manage additional entities (topics) and can potentially lead to a discovery overhead as the number of sensors scale up.
+
 Understanding Keyed Topics
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Now, let's delve into the concept of keyed topics. While traditional topics allow nodes to exchange messages freely,
 keyed topics add an additional logic of organization to this communication process.
 
-In a keyed topic, each message is associated with a **unique key**, which serves as an identifier for that message.
-This key allows nodes to differentiate between messages with similar content but different contexts or origins.
+In a keyed topic, every message is associated with a **unique key**, which serves as an identifier for that topic instance.
```

Comment:
I think this is a bit confusing. A reader could understand that each message has a unique key, when in fact all the messages belonging to the same instance have the same key

Reply:
I agree, Will rephrase it